File: PI_Integrated/Symfony Final Version/src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;


/**
 * @ORM\Entity(repositoryClass=CommandeRepository::class)
 */
class Commande
{

    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("post:read")
     */
    private $idC;

    /**
     * @ORM\Column(type="datetime")
     * @Assert\NotBlank(message="ce chanp est obligatoire")
     * @Groups("post:read")
     */
    private $date;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="ce chanp est obligatoire")
     * @Groups("post:read")
     */
    private $user;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="ce chanp est obligatoire")
     * @Groups("post:read")
     */
    private $statue = 'pending';

    /**
     * @ORM\OneToMany(targetEntity=CommandeItem::class, mappedBy="commande")
     * @Groups("post:read")
     */
    private $orderItem;

    /**
     * @ORM\OneToOne(targetEntity=Livraison::class, mappedBy="commande", cascade={"persist", "remove"})
     * @Groups("post:read")
     */
    private $livraison;

    /**
     * @ORM\Column(type="float")
     * @Groups("post:read")
     */
    private $remise = 0;

    public function __construct()
    {

    }

    public function getIdC(): ?int
    {
        return $this->idC;
    }

    public function setIdC(int $idC): self
    {
        $this->idC = $idC;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getUser(): ?string
    {
        return $this->user;
    }

    public function setUser(string $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getStatue(): ?string
    {
        return $this->statue;
    }

    public function setStatue(string $statue): self
    {
        $this->statue = $statue;

        return $this;
    }


    /**
     * @return Collection<int, CommandeItem>
     */
    public function getOrderItem(): Collection
    {
        return $this->orderItem;
    }

    public function addOrderItem(CommandeItem $orderItem): self
    {
        if (!$this->orderItem->contains($orderItem)) {
            $this->orderItem[] = $orderItem;
            $orderItem->setCommande($this);
        }

        return $this;
    }

//
    public function removeOrderItem(CommandeItem $orderItem): self
    {
        if ($this->orderItem->removeElement($orderItem)) {
            // set the owning side to null (unless already changed)
            if ($orderItem->getCommande() === $this) {
                $orderItem->setCommande(null);
            }
        }

        return $this;
    }

    public function __toString()
    {
        return $this->getIdC() . "";
    }

    public function getLivraison(): ?Livraison
    {
        return $this->livraison;
    }

    public function setLivraison(?Livraison $livraison): self
    {
        // unset the owning side of the relation if necessary
        if ($livraison === null && $this->livraison !== null) {
            $this->livraison->setCommande(null);
        }

        // set the owning side of the relation if necessary
        if ($livraison !== null && $livraison->getCommande() !== $this) {
            $livraison->setCommande($this);
        }

        $this->livraison = $livraison;

        return $this;
    }

    public function getRemise(): ?float
    {
        return $this->remise;
    }

    public function setRemise(float $remise): self
    {
        $this->remise = $remise;

        return $this;
    }

}

File: PI_Integrated/Mobile Final Version/src/com/mycompany/myapp/entities/Commande.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.myapp.entities;

import java.util.Date;

/**
 *
 * @author ashre
 */
public class Commande {
    private int id_c;
    private String date;
    private int user;
    private String statue;
    private boolean livraison;
    
 

    public boolean isLivraison() {
        return livraison;
    }

    public void setLivraison(boolean livraison) {
        this.livraison = livraison;
    }

    public Commande(int id_c, String date, int user, String statue) {
        this.id_c = id_c;
        this.date = date;
        this.user = user;
        this.statue = statue;
    }

    public int getId_c() {
        return id_c;
    }

    public void setId_c(int id_c) {
        this.id_c = id_c;
    }

    public String getDate() {
        return date;
    }

    
    public void setDate(String date) {
        this.date = date;
    }

    public int getUser() {
        return user;
    }

    public void setUser(int user) {
        this.user = user;
    }

    public String getStatue() {
        return statue;
    }

    public void setStatue(String statue) {
        this.statue = statue;
    }


    public Commande() {
    }

    
    public Commande(int id_c, String date, int user, String statue, double remise) {
        this.id_c = id_c;
        this.date = date;
        this.user = user;
        this.statue = statue;
        
    }

    @Override
    public String toString() {
        return "Commande{" + "id_c=" + id_c + ", date=" + date + ", user=" + user + ", statue=" + statue + ", livraison=" + livraison + '}';
    }

   
    
    
    
}
